web/views.py

```diff
@@ -680,34 +662,27 @@ def dispatch(self, request, *args, **kwargs):
         path replacement manually.
         """
 
-        # Manually set the language code to the default language, which isn't prefixed in URLs,
-        # to stop the locale middleware from adding a language/locale prefix back into the study URL
-        # after the request has been dispatched
-        translation.activate(settings.LANGUAGE_CODE)
-        request.LANGUAGE_CODE = settings.LANGUAGE_CODE
-
         study_uuid = kwargs.get("uuid", None)
         child_uuid = kwargs.get("child_id", None)
 
         # Check if locale (language code) is present in the URL.
         # If so, we need to re-write the request path without the locale
         # so that it points to a working study URL.
+        path = request.path
         locale_pattern = (
             rf"/(?P<locale>[a-zA-Z-].+)/studies/{study_uuid}/{child_uuid}/(?P<rest>.*?)"
         )
-        path_match = re.match(locale_pattern, request.path)
+        path_match = re.match(locale_pattern, path)
         if path_match:
-            path_no_locale = (
-                rf"/studies/{study_uuid}/{child_uuid}/{path_match.group('rest')}"
-            )
-            request.path = path_no_locale
-            request.path_info = path_no_locale
-            request.META["PATH_INFO"] = path_no_locale
+            path = f"/studies/{study_uuid}/{child_uuid}/{path_match.group('rest')}"
+            url = request.build_absolute_uri(path)
+            # Using redirect instead of super().dispatch here to get around locale/translation middleware
+            return redirect(url)
 
         if settings.DEBUG and settings.ENVIRONMENT == "develop":
             # If we're in a local environment, then redirect shortcut to switch to the ember server
-            debug_path = rf"{settings.EXPERIMENT_BASE_URL}{request.path_info}"
-            return redirect(debug_path)
-        else:
-            path = f"{study_uuid}/index.html"
-            return super().dispatch(request, path)
+            url = f"{settings.EXPERIMENT_BASE_URL}{path}"
+            return redirect(url)
+
+        path = f"{study_uuid}/index.html"
+        return super().dispatch(request, path)
```

Comment:
I don't think we should change it or put any time in researching, but do you know why the "index.html" needs to be in the URL?

Reply:
No, sorry.